### PR TITLE
Add Nicla Sense ME dashboard

### PR DESCRIPTION
**Pull request by marqdevx**
In this PR I've:
* Added inclusion of the Nicla Sense ME board (sketch)
  * Change UUID
  * Add new sensors 
  * Add callback on disconnection so the LED will turn red automatically.
* Changed the  style of the "new" dashboard
  * Add Arduino PRO logo
  * Change widget sizes, now there will be no weird spacings
  * Add snackbar with info at the top, user can remove the snackbar notification
  * Change the Pair button with the "PRO colors"
  * The board as it is smaller, the 3d preview camera is closer to the 3d model
  * Rotation of the 3D board is done with quaternions from the device
* Inside the Readme added the "new" dashboard, and linked a small tutorial explaining how to use Nicla's dashboard

Preview available at https://marqdevx.github.io/ArduinoAI/NiclaSenseME-dashboard/

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,10 +9,16 @@ This is a temporary website linking some of the demos and tools we build to supp
...
 
 ## BLE Sense Demo Dashboard
 
 [BLE Sense Demo - Arduino sketch](https://create.arduino.cc/editor/dpajak/e4128780-d76d-438f-97a4-541740cb0473/preview)
 
 [BLE Sense Demo - Web Dashboard](https://arduino.github.io/ArduinoAI/BLESense-test-dashboard) (Chrome only)
 
+## Nicla Sense ME Demo Dashboard
+
+[Nicla Sense ME BLE Demo - Arduino sketch](https://create.arduino.cc/editor/pmarquinez/b0102b0a-1d8b-4b4a-a53d-781b896d87f7/preview)
```
Comment (sebromero) [inline README.md]: @marqdevx Can we use the "official" Arduino Cloud account to host the sketch?

Comment (marqdevx) [reply README.md]: Fixed

File: NiclaSenseME-dashboard/NiclaSenseME/NiclaSenseME.ino [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+
+1) Upload this sketch to the Arduino Nano BLE sense board
+
+2) Open the following web page in the Chrome browser:
+https://arduino.github.io/ArduinoAI/NiclaSenseME-dashboard/
+
+3) Click on the green button in the web page to connect the browser to the board over BLE
+
+
+Web dashboard by D Pajak
```
Comment (sebromero) [inline NiclaSenseME-dashboard/NiclaSenseME/NiclaSenseME.ino]:
```suggestion
Web dashboard by D. Pajak
Adapted for Nicla Sense ME by P. Marquínez
```

File: NiclaSenseME-dashboard/index.html [hunk 3]
```diff
@@ -0,0 +1,752 @@
...
+  Demo showing simple dashboard of Arduino Nicla Sense ME (from the BLE Sense version)
+  
+  Based on: https://developers.google.com/web/updates/2015/07/interact-with-ble-devices-on-the-web
+  For earlier versions of Windows and Linux, you still have to go to chrome://flags/#enable-experimental-web-platform-features,
+  enable the highlighted flag, and restart Chrome for now.
+  */
+
+  var maxRecords = 64;
+
+  var BLEsense =
```
Comment (sebromero) [inline NiclaSenseME-dashboard/index.html]:
```suggestion
  var NiclaSenseME =
```

File: NiclaSenseME-dashboard/index.html [hunk 4]
```diff
@@ -0,0 +1,752 @@
...
+    BLEstatus.innerHTML = m;
+  }
+
+  //Top left pop up messaage
+  function snack(m, warningLevel) {
+    snackBar = document.getElementById("snackbar");
+    snackBar.style.visibility = "visible";
+    snackBar.className = "show";
+
+    var warnEmoji = warningLevel ? '&#x26A0  ' : '&#10004  ';
```
Comment (sebromero) [inline NiclaSenseME-dashboard/index.html]:
```suggestion
    var warnEmoji = warningLevel ? '✋  ' : '✅  ';
```

File: NiclaSenseME-dashboard/NiclaSenseME/NiclaSenseME.ino [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+  }
+
+  String address = BLE.address();
+
+  Serial.print("address = ");
+  Serial.println(address);
+
+  address.toUpperCase();
+
+  name = "BLESense-";
```
Comment (sebromero) [inline NiclaSenseME-dashboard/NiclaSenseME/NiclaSenseME.ino:94]:
```suggestion
  name = "NiclaSenseME-";
```

File: NiclaSenseME-dashboard/NiclaSenseME/NiclaSenseME.ino [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+Device sketch based on example by Sandeep Mistry and Massimo Banzi
+Sketch and web dashboard copy-fixed to be used with the Nicla Sense ME by Pablo Marquínez
+
+*/
+
+#include "Nicla_System.h"
+#include "Arduino_BHY2.h"
+#include <ArduinoBLE.h>
+
+#define BLE_SENSE_UUID(val) ("6fbe1da7-" val "-44de-92c4-bb6e04fb0212")
```
Comment (sebromero) [inline NiclaSenseME-dashboard/NiclaSenseME/NiclaSenseME.ino]: Use a different UUID for this board

File: NiclaSenseME-dashboard/BLE_spec.txt [hunk 7]
```diff
@@ -0,0 +1,116 @@
+BLE Sense - BLE specifications.txt
```
Comment (sebromero) [inline NiclaSenseME-dashboard/BLE_spec.txt:1]: Please adapt the name and the BLE IDs once you changed the UUID in the sketch.

**Review by sebromero — CHANGES_REQUESTED** [review]
Just a few things left to do. Please have a look at the comments.

**Comment by sebromero** [comment]
@marqdevx 
- For me the LED color on the board doesn't match the color on the board. 
- The LED color is not in sync when the app is loaded
- The LED light source comes from a weird place, should be the thing next to the reset button (see image)
- I suggest to add a background color to the 3D canvas. E.g. #58585A otherwise it's really hard to see.
download (2)
.

**Comment by marqdevx** [comment]
> @marqdevx
> 
>     * For me the LED color on the board doesn't match the color on the board.
> 
>     * The LED color is not in sync when the app is loaded
> 
>     * The LED light source comes from a weird place, should be the thing next to the reset button (see image)
> 
>     * I suggest to add a background color to the 3D canvas. E.g. #58585A otherwise it's really hard to see.
>       download (2)
>       .

* For the LED sync: Please check if you have the latest BHY2 library version, they changed the gbr to rgb.
* When the sketch is loaded the board is green, then if you change the Color selector it will apply the selected color to the board
* I will try to fix the light origin
* I'll try that color for the background

File: NiclaSenseME-dashboard/NiclaSenseME/NiclaSenseME.ino (hunk 2 above)
Comment (marqdevx) [reply NiclaSenseME-dashboard/NiclaSenseME/NiclaSenseME.ino]: Fixed

File: NiclaSenseME-dashboard/index.html (hunk 4 above)
Comment (marqdevx) [reply NiclaSenseME-dashboard/index.html]: Fixed

File: NiclaSenseME-dashboard/NiclaSenseME/NiclaSenseME.ino (hunk 5 above)
Comment (marqdevx) [reply NiclaSenseME-dashboard/NiclaSenseME/NiclaSenseME.ino:94]: Fixed

File: NiclaSenseME-dashboard/NiclaSenseME/NiclaSenseME.ino (hunk 6 above)
Comment (marqdevx) [reply NiclaSenseME-dashboard/NiclaSenseME/NiclaSenseME.ino]: Fixed

File: NiclaSenseME-dashboard/BLE_spec.txt (hunk 7 above)
Comment (marqdevx) [reply NiclaSenseME-dashboard/BLE_spec.txt:1]: Fixed

**Comment by marqdevx** [comment]
@sebromero 
Everything added/fixed

New stuff:
* Now the dashboard will appear with the LED on green.
* On connect, the LED will be blue (both color picker and the device)
* When the device gets disconnected, the LED will turn red (both web dashboard and the board)
* Camera is closer to the board (as it is smaller than the Nano BLE Sense)

image

File: NiclaSenseME-dashboard/index.html [hunk 8]
```diff
@@ -0,0 +1,772 @@
...
+  </style>
+</head>
+
+<body>
+
+  <div class="container">
+
+    <div id="snackbar"></div>
+    <div class="status widget">
+      <button id="pairButton">PAIR NICLA</button>
```
Comment (sebromero) [inline NiclaSenseME-dashboard/index.html]:
```suggestion
      <button id="pairButton">CONNECT BOARD</button>
```

File: NiclaSenseME-dashboard/index.html [hunk 9]
```diff
@@ -0,0 +1,772 @@
...
+      <div class="label statusBar" id="bluetooth">Click the pair button to connect your device</div>
```
Comment (sebromero) [inline NiclaSenseME-dashboard/index.html]:
```suggestion
      <div class="label statusBar" id="bluetooth">Click the connect button to connect your device</div>
```

**Review by sebromero — CHANGES_REQUESTED** [review]
I'd rename the button. "Nicla" is the name of a whole product family so I'd try to avoid using it.

File: NiclaSenseME-dashboard/index.html (hunk 8 above)
Comment (marqdevx) [reply NiclaSenseME-dashboard/index.html]: What about only "CONNECT"

Comment (sebromero) [reply NiclaSenseME-dashboard/index.html]: > What about only "CONNECT"

Works too 👍

**Pull request merged**